### DISCUSSION TITLE
hot to fallback to default fallback of matplotlib

### DISCUSSION THREAD

**Discussion by bassamsdata** [benlubas/molten-nvim#168] [Q&A] [answered]
Hello, 
thank you in advance for any suggestions here.

I'm reaching out to inquire if there's a way to adjust settings to make the default action of the `plt.show() `function in matplotlib function as expected.

Normally, in a `Python` or `IPython` environment, executing this command in the terminal should automatically open the result in a browser. However, with my current setup, regardless of the options I try in molten, this isn't happening.

The issue arose after updating my packages on my Mac. Specifically,` image.nvim` stopped functioning properly; it seems unable to recognize `imagerock` anymore. I attempted to resolve this by reinstalling both `imagerock` and `magick` via Lua 5.1, but unfortunately, I've had no success, even after manually adding the path.

For reference, here is the [source fallback](https://matplotlib.org/stable/users/explain/figure/backends.html#the-builtin-backends) for matplotlib.

here's a small example:

```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots()

fruits = ['apple', 'blueberry', 'cherry', 'orange']
counts = [40, 100, 30, 55]
bar_labels = ['red', 'blue', '_red', 'orange']
bar_colors = ['tab:red', 'tab:blue', 'tab:red', 'tab:orange']

ax.bar(fruits, counts, label=bar_labels, color=bar_colors)

ax.set_ylabel('fruit supply')
ax.set_title('Fruit supply by kind and color')
ax.legend(title='Fruit color')

plt.show()
```

**another question** 
Is it possible to display virtual text within `.py` files, akin to how it functions within `quarto` or `jupyter` notebooks? Presently, in plain Python text files, only the popup version appears, not the virtual text.

here is the settings:
```lua
vim.g.molten_auto_open_output = false
vim.g.molten_virt_text_output = true
vim.g.molten_virt_lines_off_by_1 = false
vim.g.molten_wrap_output = true
```

currently it work like this:
<img width="744" alt="Screenshot 2024-03-19 at 11 30 34 AM" src="https://github.com/benlubas/molten-nvim/assets/105807570/f3379cab-ff3e-4e13-814c-51e9aaa76cd0">

**Comment by benlubas** [comment, marked as answer]
> Is it possible to display virtual text within .py files, akin to how it functions within quarto or jupyter notebooks? Presently, in plain Python text files, only the popup version appears, not the virtual text.

Yes. With just the settings you have listed you should see virtual text everywhere. I suspect that you've copied a bit of code from [these docs](https://github.com/benlubas/molten-nvim/blob/main/docs/Notebook-Setup.md#change-molten-settings-based-on-filetype) which toggles some settings when you enter a `.py` file. So just removing that bit of code should let it work the way you want.

---

For the matplot lib thing, molten currently behaves the same way that Jupyter Lab does. Both apps just render the image.

![image](https://github.com/benlubas/molten-nvim/assets/56943754/c0ea9d56-1818-4ca7-86b4-6b8c80912a54)

If you want to see images in a browser, you can do that with the `:MoltenImagePopup` command and `vim.g.molten_auto_image_popup` config option. Both documented in the readme.

I'll note that command will open your images in your operating systems default image viewer, which might not necessarily be your browser. But hopefully that is still good enough.

**Reply by bassamsdata** [reply]
Thank you, I thought I removed the .py filetype thing but it looks like I didn't. thanks, all work now.
